src/router/validation/subscriptionValidator.test.ts

```diff
@@ -0,0 +1,202 @@
+const singleTenantValidator = new SubscriptionValidator(search, singleTenantAllowList, false);
+
+describe.each([
+    ['multi-tenancy mode', multiTenantValidator, 'https://fake-end-point-tenant1', 'tenant1'],
+    ['single-tenancy mode', singleTenantValidator, 'https://fake-end-point-2', undefined],
+])(
+    'Valid resources in %s',
+    (testName: string, validator: SubscriptionValidator, endpoint: string, tenantId: string | undefined) => {
+        test('No error when validating valid Subscription resource', async () => {
+            const subscription = getSubscriptionResource(endpoint);
+            await expect(validator.validate(subscription, tenantId)).resolves.toEqual(undefined);
+        });
+
+        test('No error when validating valid Bundle resource that contains valid Subscription resource', async () => {
+            const subscription = getSubscriptionResource(endpoint);
+            const bundle = getBundleResource(subscription);
+            await expect(validator.validate(bundle, tenantId)).resolves.toEqual(undefined);
+        });
+
+        test('No error when validating resources that are not Subscription or Bundle', async () => {
+            await expect(validator.validate(invalidPatient, tenantId)).resolves.toEqual(undefined);
+        });
+
+        test('No error when validating Bundle that does not contain Subscription resource', async () => {
+            const bundle = getBundleResource(validPatient);
+            await expect(validator.validate(bundle, tenantId)).resolves.toEqual(undefined);
+        });
+    },
+);
+
+describe.each([
+    ['multi-tenancy mode', multiTenantValidator, 'https://fake-end-point-tenant2', 'tenant2'],
+    ['single-tenancy mode', singleTenantValidator, 'https://fake-end-point-1', undefined],
+])(
+    'Invalid resource in %s',
+    (testName: string, validator: SubscriptionValidator, endpoint: string, tenantId: string | undefined) => {
+        test('Show error when status is not "requested" or "off"', async () => {
+            const subscription = getSubscriptionResource(endpoint);
+            subscription.status = 'active';
+            await expect(validator.validate(subscription, tenantId)).rejects.toThrowError(
+                new InvalidResourceError(
+                    'Subscription resource is not valid. Error was: data.status should be equal to one of the allowed values',
+                ),
+            );
+        });
+
+        test('Show error when payload is not application/fhir+json', async () => {
+            const subscription = getSubscriptionResource(endpoint);
+            subscription.channel.payload = 'application/xml';
+            await expect(validator.validate(subscription, tenantId)).rejects.toThrowError(
+                new InvalidResourceError(
+                    'Subscription resource is not valid. Error was: data.channel.payload should be equal to constant',
```

Comment:
you can use `ajv-errors` and the `errorMessage` field to have better error messages.

e.g. https://github.com/awslabs/fhir-works-on-aws-routing/blob/836719f7bdc4f424ad7173695cf8a8b96dcb7d3f/src/operationDefinitions/USCoreDocRef/parseParams.ts#L83

the ajv default error of "should be equal to constant" is not great for customers